### ISSUE TITLE
feature: add standard set of "ambient" props to context

### ISSUE THREAD

**Issue by toddbaert** [open-feature/flagd#851]
We already need to add the `flag_key` to the context to improve out bucketing algorithm, (see [issue](https://github.com/open-feature/flagd/issues/843) and [PR](https://github.com/open-feature/flagd/pull/847)).

We should consider what other properties could be useful, and add them under some sort of namespace to the context. A list could include:

- `flag_key`
- `current_timestamp`

I kept thinking about things like `version` (the version of flagd) or `connection_type` (HTTP vs gRPC), but these would be unreliable and inconsistent between flagd RPC and in-process implementations. We should come up with a list that's relevant to all models, it think.

**Comment by AHB102** [comment]
I am new to open source, can I try to resolve this issue ?

**Comment by toddbaert** [comment]
> I am new to open source, can I try to resolve this issue ?

Definitely. I would start by first monitoring the progress on the [blocking issue](https://github.com/open-feature/flagd/pull/847), trying to understand those changes, and helping with review.

**Comment by AHB102** [comment]
Definitely! I will look into it.

**Comment by toddbaert** [comment]
@AHB102 Are you still interested in doing this? I can assign to you if you'd like. The pre-requisite work is done.

**Comment by AHB102** [comment]
> @AHB102 Are you still interested in doing this? I can assign to you if you'd like. The pre-requisite work is done.

Currently working on something else,I'm sorry I can't work on this.

**Comment by toddbaert** [comment]
> > @AHB102 Are you still interested in doing this? I can assign to you if you'd like. The pre-requisite work is done.
> 
> Currently working on something else,I'm sorry I can't work on this.

I understand, no problem!

**Comment by craigpastro** [comment]
Hi @toddbaert. I can finish this off if that is okay. Are you still okay with the properties listed above?

**Comment by beeme1mr** [comment]
I think we should keep it simple and add more as new use cases emerge. My suggestion would be to start with `timestamp` for now. That would allow us to create time based flag configurations in a constant way.

In my opinion, we don't need to include `current` in the name. We can include a full description in the docs.

**Comment by craigpastro** [comment]
Thank you @beeme1mr. Sounds good. I'll add `timestamp` for now.

**Comment by craigpastro** [comment]
By the way @toddbaert, would you like to close this issue for now and create a new one if you think of other properties to add?

**Comment by toddbaert** [comment]
> By the way @toddbaert, would you like to close this issue for now and create a new one if you think of other properties to add?

Ya I think we can close this now. `$flag.flagKey` and `$flagd.timestamp` are both clearly helpful, but I can't think of others and haven't heard of any. The only thing I suspect might be helpful is the ability to specify custom values on the cmdline and add them:

`flagd start ... --add-prop myProp=123` would add `$flagd.myProp` to the context with value `"123"`.

_**BUT**_ I'd really not like to add this functionality unless there becomes a clear need for it.
